### PR TITLE
docs(generate): use swagger-js-codegen to generate markdown docs

### PR DESCRIPTION
**Pull request by silasbw**
Include mustache templates and markdown for the included swagger specs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,9 @@
   "scripts": {
     "check-coverage": "istanbul check-coverage",
     "coverage": "istanbul cover ./node_modules/.bin/_mocha && istanbul report cobertura",
+    "docs": "for v in 1.7 1.8 1.9; do scripts/docs.js --spec lib/specs/swagger-$v.json.gz --output docs/$v.md; done",
```
Comment (gisenberg) [inline package.json]: This won't work cross-platform.

**Comment by jcrugzz** [comment]
@silasbw these should probably be in `.npmignore`

File: package.json (hunk 1 above)
Comment (jeffpearce) [reply package.json]: For my own learning - which part will break? I'm assuming on Windows?

File: scripts/docs.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  alphanumeric: function () {
+    return function (text, render) {
+      return render(text).replace(/\W/g, '');
+    };
+  },
+
+  /**
+   * Replace newline characters in rendered text with <br/>s.
+   * @returns {funtion} function
```
Comment (jeffpearce) [inline scripts/docs.js]: nit: `{function}`

File: scripts/docs.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  markdownBreaks: function () {
+    return function (text, render) {
+      return render(text).replace(/\n/g, '<br/>');
```
Comment (jeffpearce) [inline scripts/docs.js]: Fo we need to support CRLF for Windows?

Comment (gisenberg) [reply scripts/docs.js]: It might be worth calling out these line endings specifically in `.gitattributes`

File: scripts/docs.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    };
+  },
+
+  /**
+   * When in a method section, return the full kubernetes-client name.
+   * @returns {funtion} function
```
Comment (jeffpearce) [inline scripts/docs.js]: nit: `{function}` again

File: package.json (hunk 1 above)
Comment (gisenberg) [reply package.json]: 1. bash-style for-loop syntax won't work on Windows (syntax error)
1. executing scripts/docs.js directly won't work on Windows (not an executable)

**Comment by silasbw** [comment]
@jcrugzz, we're declaring the directories / files that need to be included (instead of .npmignoring things):

https://github.com/godaddy/kubernetes-client/blob/master/package.json#L28

**Pull request merged**